torch/_inductor/codegen/cpp_wrapper_cpu.py

```diff
@@ -1459,11 +1471,71 @@ def codegen_multi_output(self, name, value):
         if not config.abi_compatible:
             super().codegen_multi_output(name, value)
 
-    def codegen_subgraph(self, subgraph, outer_inputs, outer_outputs):
-        raise NotImplementedError("Control flow NYI in C++ wrapper codegen.")
+    def codegen_subgraph_prefix(self, subgraph, outer_inputs, outer_outputs):
+        for inner_input, outer_input in zip(subgraph.graph.graph_inputs, outer_inputs):
+            if config.abi_compatible:
+                # in ABI-compatible mode, we copy the underlying at::Tensor of the conditional
+                # input (outer_input) into another at::Tensor to be used as a subgraph input
+                # (inner_input) in the nested scope. we can't std::move here, as the codegened
+                # outer input may be an expression / rvalue (e.g., reinterpret_view(x)), so we
+                # can't necessarily std::move it back to the origin (x).
```

Comment:
It seems that we might be able to leverage ```shared_ptr``` here, but I think it would require fair amount of work to pave the foundations.